README.md
medicalrisk
======

medicalrisk is a collection of tools and data for performing patient-level medical 
risk adjustment using administrative data.

The data sets include `icd9cm_charlson` and `icd9cm_elixhauser`, which are data
frames that map ICD-9-CM codes to comorbidities, using the Charlson
and Elixhauser classifications respectively.  The mapping comes from the [paper by 
Quan in 2005](http://www.ncbi.nlm.nih.gov/pubmed/16224307), and uses 
the [CMS ICD-9-CM table.](https://www.cms.gov/Medicare/Coding/ICD9ProviderDiagnosticCodes/codes.html)  
With a properly formatted source data frame of patients and ICD-9-CM values, 
the `cast` or `dcast` methods can be used to generate comorbidity tables.

The `add_charlson_index_df` method will calculate the Charlson Comorbidity Index 
for each patient.

For data sources that present ICD-9-CM codes as a comma-separated listing, the 
`melt_icd9list` method will convert such data tables into a one-row-per-code format
suitable for `join` or `merge` to the `icd9cm_*` tables.

Data Sources
------

For implementation, testing, and examples several external data sources are used:

* [ICD-9-CM code list from CMS](https://www.cms.gov/Medicare/Coding/ICD9ProviderDiagnosticCodes/codes.html)
* [Vermont Uniform Hospital Discharge Data Set](http://healthvermont.gov/research/hospital-utilization/RECENT_PU_FILES.aspx)
* [Risk Stratification Index](http://my.clevelandclinic.org/anesthesiology/outcomes-research/risk-stratification-index.aspx)


NEWS
medicalrisk v1.2 (release date 2016-01-23)
==============

* changed external data code to pull datafiles from source instead of including them in tarball
* changes to accommodate github release
* added "advanced" vignette

medicalrisk v1.1 (release date 2015-01-06)
==============

* first CRAN release
* fixed various small issues to pass CRAN testing

medicalrisk v1.0 (internal release)
==============

* Initial release


tests/test-all.R
library(testthat)
library(medicalrisk)

test_package("medicalrisk")


inst/tests/test-melt_icd9list.R
test_that("empty lists return empty df", {
    cases <- data.frame(id=c(), icd9list=c())
    expect_that(melt_icd9list(cases, "id", "icd9list"), equals(data.frame()))
})

test_that("numeric input to icd9list", {
    cases <- data.frame(id=c(1,2), icd9list=c(100.1,200.1))
    expect_that(melt_icd9list(cases, "id", "icd9list"), 
                equals(data.frame(id=c(1,2), icd9cm=c("1001","2001"))))
})

test_that("real icd9 codes", {
    cases <- data.frame(id=c(1,2), icd9list=c('162.4,070.30,155.0,401.9','996.52,E878.8,V45.86'))
    expect_that(melt_icd9list(cases, "id", "icd9list"), 
                equals(data.frame(id=c(1, 1, 1, 1, 2, 2, 2), 
                                  icd9cm=c("1624", "07030", "1550", "4019", "99652", "E8788", "V4586"))))
})


R/calculate-rsi.R
# Code for calculating Sessler RSI

#' Returns the covariate coefficient for a particular diagnosis
#' or procedure code, along with the actual code that was found in
#' the internal database of coefficients. If a child code is supplied
#' but its parent is in the database, the coefficient for the parent
#' will be returned, along with that parent code. For example, if
#' D1231 is supplied but only D123 is available, D123 will be used.
#' This is apparently how the SPSS sample code works.
#' 
#' @param code A single ICD-9-CM code
#' @param betalist One of the rsi_beta_* datasets (supplied with this package)
#' @return Covariate coefficient. You must sum all of these for a given patient and then
#'     subtract the appropriate population beta (e.g. rsi_beta_1yrpod$popbeta)
#' @importFrom hash has.key
#' @examples
#' # get coefficient for hypercholesterolemia
#' sessler_get_single_beta('D2720', rsi_beta_inhosp)
#' # Also works with extra 0 on the end
#' sessler_get_single_beta('D27200', rsi_beta_inhosp)
#' @export
sessler_get_single_beta <- function(code, betalist) {
  code <- as.character(code)
  if(nchar(code) < 4) return(list(coeff=0, code=NULL)) # Sanity check
  
  # Actually find the entry. For some reason R does not have
  # real hashes built in, so we have to resort to the 'hash'
  # add-on package available in CRAN.
  if(has.key(code, betalist)) {
    return(list(coeff=betalist[[code]], code=code))
  } else {
    # Didn't find a match? Perhaps only this code's parent is in the
    # betalist. Chop off a character and try again. We do this down
    # to 4 characters.
    if(nchar(code) > 4) {
      code <- substr(code, 1, nchar(code)-1)
      return(sessler_get_single_beta(code, betalist))
    }
    # Couldn't find it, so give up and return zero
    return(list(coeff=0, code=NULL))
  }
}

#' Returns composite Sessler risk stratification index, given a list of ICD-9-CM codes.
#'
#' ICD-9-CM codes must have periods removed.  Diagnostic codes are prefixed with 
#' 'D' while procedure codes are prefixed with 'P'. So, diagnostic code
#' \code{404.03} should be \code{"D40403"}.
#' 
#' Note: A subsequent publication (Sigakis, 2013) found the following:
#' "Calibration "in-the-large" for RSI in-hospitalmortality illustrated a discrepancy 
#' between actual (1.5%) and predicted (51.7%) in-hospital mortality. The authors 
#' identified a regression constant (-2.198) in the published RSI 
#' "all-covariates.xls" file that was not used in the published SPSS implementation 
#' macro."
#' 
#' @author Tom Joseph <thomas.joseph@@mountsinai.org>, 
#' Patrick McCormick <patrick.mccormick@@mountsinai.org>
#' @param icd9 a unique character vector of ICD-9-CM codes
#' @return The risk stratification index score
#' @references 1. Sessler DI, Sigl JC, Manberg PJ, Kelley SD, Schubert A, Chamoun NG.
#'     Broadly applicable risk stratification system for predicting duration of hospitalization and mortality.
#'     Anesthesiology. 2010 Nov;113(5):1026-37. doi: 10.1097/ALN.0b013e3181f79a8d.   
#' 
#' 2. Sigakis MJG, Bittner EA, Wanderer JP: Validation of a risk stratification 
#'     index and risk quantification index for predicting patient outcomes: 
#'     in-hospital mortality, 30-day mortality, 1-year mortality, and length-of-stay. 
#'     Anesthesiology 2013; 119:525-40
#' @examples
#' # Calculate RSI for each patient ("id") in dataframe
#' cases <- data.frame(id=c(1,1,1,2,2,2),
#'   icd9cm=c("D20206","D24220","D4439","D5064","DE8788","D40403"))
#' library(plyr)
#' ddply(cases, .(id), function(x) { icd9cm_sessler_rsi(x$icd9cm) } )
#' @export
icd9cm_sessler_rsi <- function(icd9) {
  icdlist <- unique(icd9)    
  data.frame(
    rsi_1yrpod = sum(sapply(sapply(icdlist, sessler_get_single_beta, medicalrisk::rsi_beta_1yrpod)['coeff',], function(x) x[[1]])) - medicalrisk::rsi_beta_1yrpod$popbeta,
    rsi_30dlos = sum(sapply(sapply(icdlist, sessler_get_single_beta, medicalrisk::rsi_beta_30dlos)['coeff',], function(x) x[[1]])) - medicalrisk::rsi_beta_30dlos$popbeta,
    rsi_30dpod = sum(sapply(sapply(icdlist, sessler_get_single_beta, medicalrisk::rsi_beta_30dpod)['coeff',], function(x) x[[1]])) - medicalrisk::rsi_beta_30dpod$popbeta,
    rsi_inhosp = sum(sapply(sapply(icdlist, sessler_get_single_beta, medicalrisk::rsi_beta_inhosp)['coeff',], function(x) x[[1]])) - medicalrisk::rsi_beta_inhosp$popbeta
  ) 
}

#' Validates this Sessler RSI implementation against reference data
#' 
#' Requires that "sample data rev2.csv" and "sample results rev2.csv" be available 
#' in datasrc directory
#'
#' @author Patrick McCormick <patrick.mccormick@@mountsinai.org>
#' @return Table of patients with scores >0.001 difference.
#' @seealso rsi_sample_data, rsi_sample_results
#' @references 1. Sessler DI, Sigl JC, Manberg PJ, Kelley SD, Schubert A, Chamoun NG.
#'     Broadly applicable risk stratification system for predicting duration of hospitalization and mortality.
#'     Anesthesiology. 2010 Nov;113(5):1026-37. doi: 10.1097/ALN.0b013e3181f79a8d.
#' @importFrom plyr ddply
#' @importFrom reshape2 melt
#' @examples
#' \dontrun{
#' verify_sessler_rsi()
#' }
#' @export
verify_sessler_rsi <- function() {
  sessler_sample_melted <- melt(medicalrisk::rsi_sample_data, c("patient_id"), value.name="icd9cm")
  sessler_medicalrisk_results <- ddply(sessler_sample_melted, "patient_id", function(x) { icd9cm_sessler_rsi(x$icd9cm) })
  test_grid <- merge(sessler_medicalrisk_results,
                     medicalrisk::rsi_sample_results[,c('patient_id','RSI_LOS','RSI_INHOSP','RSI_30Day','RSI_1YR')])
  test_grid$diff_1yrpod <- test_grid$rsi_1yrpod - test_grid$RSI_1YR
  test_grid$diff_30dlos <- test_grid$rsi_30dlos - test_grid$RSI_LOS
  test_grid$diff_30dpod <- test_grid$rsi_30dpod - test_grid$RSI_30Day
  test_grid$diff_inhosp <- test_grid$rsi_inhosp - test_grid$RSI_INHOSP
  test_grid$diff_sum <- test_grid$diff_1yrpod + test_grid$diff_30dlos + test_grid$diff_30dpod + test_grid$diff_inhosp
  test_grid$diff_problem <- test_grid$diff_sum > 0.0001
  
  test_grid[test_grid$diff_problem,]
}


datasrc/generate_data.R
# Script to generate files in "data" subdirectory from files in "datasrc" subdirectory
# Run this before running R CMD Check
# 
# $Id: generate_data.R 4008 2016-01-23 20:59:21Z patrick $

library(hash)
library(reshape2)

# make sure data dir exists, create if not
if (!file_test("-d","data")) {
  if (file.exists("data")) {
    stop("file \"data\" exists, unable to make directory \"data\"")
  }
  if (!dir.create("data")) {
    stop("failed attempting to create directory \"data\"")
  }
}

# icd9cm_list is a list of all ICD-9-CM diagnostic and procedure codes
# diagnostic codes are prefixed with "D", procedure codes prefixed with "P"
# Retrieved from CMS URL below

cms_icd9_zip_url <- 'https://www.cms.gov/Medicare/Coding/ICD9ProviderDiagnosticCodes/Downloads/ICD-9-CM-v32-master-descriptions.zip'
cms_icd9_zip <- 'datasrc/icd9cm.zip'
cms_icd9_dx_file <- 'datasrc/CMS32_DESC_SHORT_DX.txt'
cms_icd9_px_file <- 'datasrc/CMS32_DESC_SHORT_SG.txt'
cms_icd9_files <- c(cms_icd9_dx_file, cms_icd9_px_file)

icd9cm_list <- (function(){
  if(!all(file.exists(cms_icd9_files))) {
    if(!file.exists(cms_icd9_zip)) {
      download.file(cms_icd9_zip_url, cms_icd9_zip)
    }
    unzip(cms_icd9_zip, files = basename(cms_icd9_files), exdir = 'datasrc')
  }
  
  f <- read.fwf(cms_icd9_dx_file, widths=c(5,255), colClasses="character", strip.white=T)
  dx <- paste0(rep.int('D',length(f$V1)), f$V1)
  
  f <- read.fwf(cms_icd9_px_file, widths=c(4,255), colClasses="character", strip.white=T)
  px <- paste0(rep.int('P',length(f$V1)), f$V1)
  
  c(dx,px)
})()

charlson_list <- list(
  mi = "Myocardial infarction",
  chf = "Congestive heart failure",
  perivasc = "Peripheral vascular disorders",
  cvd = "Cerebrovascular disease",
  dementia = "Dementia",
  chrnlung = "Chronic pulmonary disease",
  rheum = "Rheumatic disease",
  ulcer = "Peptic ulcer disease",
  liver = "Mild liver disease",
  dm = "Diabetes without chronic complication",
  dmcx = "Diabetes with chronic complication",
  para = "Hemiplegia or paraplegia",
  renal = "Renal disease",
  tumor = "Any malignancy, including lymphoma and leukemia, except malignant neoplasm of skin",
  modliver = "Moderate or severe liver disease",
  mets = "Metastatic solid tumor",
  aids = "AIDS/HIV"
)

elixhauser_list <- list(
  chf = "Congestive heart failure",
  arrhythmia = "Cardiac arrhythmias",
  valve = "Valvular disease",
  pulmcirc = "Pulmonary circulation disorders",
  perivasc = "Peripheral vascular disorders",
  htn = "Hypertension, uncomplicated",
  htncx = "Hypertension, complicated",
  para = "Paralysis",
  neuro = "Other neurological disorders",
  chrnlung = "Chronic pulmonary disease",
  dm = "Diabetes, uncomplicated",
  dmcx = "Diabetes, complicated",
  hypothy = "Hypothyroidism",
  renlfail = "Renal failure",
  liver = "Liver disease",
  ulcer = "Peptic ulcer disease excluding bleeding",
  aids = "AIDS/HIV",
  lymph = "Lymphoma",
  mets = "Metastatic Cancer",
  tumor = "Solid tumor without metastasis",
  rheum = "Rheumatoid arthritis/Collagen vascular diseases",
  coag = "Coagulopathy",
  obese = "Obesity",
  wghtloss = "Weight loss",
  lytes = "Fluid and electrolyte disorders",
  bldloss = "Blood loss anemia",
  anemdef = "Deficiency anemia",
  alcohol = "Alcohol abuse",
  drug = "Drug abuse",
  psych = "Psychoses",
  depress = "Depression"
)
  
rcri_list <- list(
  chf = "Congestive heart failure",
  cvd = "Cerebrovascular disease",
  dm = "Diabetes",
  ischemia = "Ischemic heart disease",
  renlfail = "Renal failure"
)

# Weights
# see data.R for details
charlson_weights_orig <- list(
  mi = 1,
  chf = 1,
  perivasc = 1,
  cvd = 1,
  dementia = 1,
  chrnlung = 1,
  rheum = 1,
  ulcer = 1,
  liver = 1,
  dm = 1,
  dmcx = 2,
  para = 2,
  renal = 2,
  tumor = 2,
  modliver = 3,
  mets = 6,
  aids = 6
)

# see data.R for details
charlson_weights <- list(
  mi = 1,
  chf = 2,
  perivasc = 1,
  cvd = 1,
  dementia = 3,
  chrnlung = 2,
  rheum = 0,
  ulcer = 0,
  liver = 2,
  dm = 1,
  dmcx = 2,
  para = 1,
  renal = 3,
  tumor = 2,
  modliver = 4,
  mets = 6,
  aids = 4
)

# VT Inpatient Sample
# see data.R for details

vt_inp_sample <- (function() {
  df <- read.csv("datasrc/vt_top100.csv", stringsAsFactors=FALSE)

  names(df) <- tolower(names(df))
  df$scu_days[is.na(df$scu_days)] <- 0
  df_melt <- melt(df[,c(1,11:33,54:56)], id.vars=c("id","scu_days","drg","mdc"), na.rm=T,
                    variable.name="dx", value.name="icd9cm")
  df_px_melt <- melt(df[,c(1,11:13,34:53)], id.vars=c("id","scu_days","drg","mdc"), na.rm=T, 
                     variable.name="px", value.name="icd9cm")
  cols <- intersect(names(df_melt), names(df_px_melt))
  df_melt <- df_melt[nchar(df_melt$icd9cm) > 0, cols]
  df_melt$icd9cm <- paste('D', df_melt$icd9cm, sep='') 
  df_px_melt <- df_px_melt[nchar(df_px_melt$icd9cm) > 0, cols]
  df_px_melt$icd9cm <- paste('P', df_px_melt$icd9cm, sep='') 
  df_melt <- rbind(df_melt, df_px_melt)
  df_melt$icd9cm <- factor(df_melt$icd9cm)
  df_sorted <- df_melt[order(df_melt$id),]
  row.names(df_sorted) <- NULL
  df_sorted
})()

rsi_zip_url <- 'http://my.clevelandclinic.org/Documents/anesthesiology/RSI-calculation.zip'
rsi_zip <- 'datasrc/rsi.zip'
rsi_inhosp_csv <- 'datasrc/All Proc - Step 4 - INHOSP - May-20-10.csv'
rsi_30dpod_csv <- 'datasrc/All Proc - Step 4 - 30DPOD - May-20-10.csv'
rsi_1yrpod_csv <- 'datasrc/All Proc - Step 4 - 1YRPOD - May-20-10.csv'
rsi_30dlos_csv <- 'datasrc/All Proc - Step 4 - 30DLOS - May-20-10.csv'
rsi_sampledata_csv <- 'datasrc/sample data rev2.csv'
rsi_sampleresults_csv <- 'datasrc/sample results rev2.csv'
rsi_files <- c(rsi_inhosp_csv, rsi_30dpod_csv, rsi_1yrpod_csv, rsi_30dlos_csv, rsi_sampledata_csv, rsi_sampleresults_csv)

# RSI files
(function(){
  if(!all(file.exists(rsi_zip))) {
    if(!file.exists(rsi_zip)) {
      download.file(rsi_zip_url, rsi_zip)
    }
    unzip(rsi_zip, files = basename(rsi_files), exdir = 'datasrc')
  }

  csv_inhosp <- read.csv(rsi_inhosp_csv, header=TRUE, strip.white=TRUE)
  csv_30dpod <- read.csv(rsi_30dpod_csv, header=TRUE, strip.white=TRUE)
  csv_1yrpod <- read.csv(rsi_1yrpod_csv, header=TRUE, strip.white=TRUE)
  csv_30dlos <- read.csv(rsi_30dlos_csv, header=TRUE, strip.white=TRUE)

  rsi_beta_inhosp <- hash(keys=csv_inhosp$dp_code_cox, values=csv_inhosp$beta)
  rsi_beta_30dpod <- hash(keys=csv_30dpod$dp_code_cox, values=csv_30dpod$beta)
  rsi_beta_1yrpod <- hash(keys=csv_1yrpod$dp_code_cox, values=csv_1yrpod$beta)
  rsi_beta_30dlos <- hash(keys=csv_30dlos$dp_code_cox, values=csv_30dlos$beta)

  rsi_sample_data <- read.csv(rsi_sampledata_csv)
  rsi_sample_results <- read.csv(rsi_sampleresults_csv)

  # For some reason the population beta, where they apply, are repeated for every row
  rsi_beta_inhosp$popbeta <- 0
  rsi_beta_30dpod$popbeta <- csv_30dpod[1,'PopBeta']
  rsi_beta_1yrpod$popbeta <- csv_1yrpod[1,'PopBeta']
  rsi_beta_30dlos$popbeta <- csv_30dlos[1,'PopBeta']
  
  # create RDA files
  save(rsi_beta_inhosp, file="data/rsi_beta_inhosp.rda", compress=TRUE)
  save(rsi_beta_30dpod, file="data/rsi_beta_30dpod.rda", compress=TRUE)
  save(rsi_beta_1yrpod, file="data/rsi_beta_1yrpod.rda", compress=TRUE)
  save(rsi_beta_30dlos, file="data/rsi_beta_30dlos.rda", compress=TRUE)
  save(rsi_sample_data, file="data/rsi_sample_data.rda", compress=TRUE)
  save(rsi_sample_results, file="data/rsi_sample_results.rda", compress=TRUE)
})()

# create RDA files
save(icd9cm_list, file="data/icd9cm_list.rda", compress=TRUE)
save(charlson_list, file="data/charlson_list.rda", compress=TRUE)
save(elixhauser_list, file="data/elixhauser_list.rda", compress=TRUE)
save(charlson_weights_orig, file="data/charlson_weights_orig.rda", compress=TRUE)
save(charlson_weights, file="data/charlson_weights.rda", compress=TRUE)
save(vt_inp_sample, file="data/vt_inp_sample.rda", compress=TRUE)


R/data.R
#' List of Charlson comorbidities
#' 
#' @docType data
#' @name charlson_list
#' @usage charlson_list
#' @format A list, with one column for each comorbidity; value is a textual description 
#' @keywords datasets
#' @examples
#' # List the strings used to refer to Charlson comorbidities
#' names(charlson_list)
#' 
#' # List descriptions of comorbidities
#' charlson_list
NULL

#' List of Elixhauser comorbidities
#' 
#' @docType data
#' @name elixhauser_list
#' @usage elixhauser_list
#' @format A list, with one column for each comorbidity; value is a textual description 
#' @keywords datasets
#' @examples
#' # List the strings used to refer to Elixhauser comorbidities
#' names(elixhauser_list)
#' 
#' # List descriptions of comorbidities
#' elixhauser_list
NULL

#' List of ICD-9-CM diagnostic and procedural codes
#' 
#' ICD-9-CM codes have the periods removed.  Diagnostic codes are prefixed with 
#' 'D' while procedure codes are prefixed with 'P'. So, diagnostic code
#' \code{404.03} appears as \code{"D40403"}.
#' 
#' Obsolete codes not active in 2012 are not present, and may cause this dataset
#' to miss certain classifications when applied to older datasets.  For example, 
#' codes 043 and 044 (both obsolete AIDS codes) are not included.
#' 
#' @docType data
#' @name icd9cm_list
#' @usage icd9cm_list
#' @format A string vector
#' @references 1. \url{https://www.cms.gov/Medicare/Coding/ICD9ProviderDiagnosticCodes/codes.html}
#'   
#' @keywords datasets
#' @examples
#' # Count procedural codes
#' length(icd9cm_list[grep('^P',icd9cm_list)])
NULL

#' Map of Charlson comorbidity categories to weights
#' 
#' List that links the Charlson comorbidity categories to the original 
#' weights (specified in the original Charlson paper, Table 3)
#' 
#' Original Weights:
#' 
#' 1 = MI, CHF, PVD, CVD, Dementia, Chronic pulm dz, Connective tissue dz,
#' Ulcer, Mild liver dz, Diabetes
#' 
#' 2 = Hemiplegia, Mod or severe renal dz, Diabetes with end organ damage, Any
#' tumor, Leukemia, Lymphoma
#' 
#' 3 = Moderate or severe liver dz
#' 
#' 6 = Metastatic solid tumor, AIDS
#' 
#' @docType data
#' @name charlson_weights_orig
#' @usage charlson_weights_orig
#' @format A list, with Charlson comorbidities as names and weight as value
#' @references 1. Charlson ME, Pompei P, Ales KL, MacKenzie CR: A new method of
#'   classifying prognostic comorbidity in longitudinal studies: development and
#'   validation. Journal of chronic diseases 1987; 40:373-83 
#'   \url{http://www.ncbi.nlm.nih.gov/pubmed/3558716} 
#' @seealso \code{\link{charlson_weights}},
#' \code{\link{icd9cm_charlson_deyo}}, 
#' \code{\link{icd9cm_charlson_romano}},
#' \code{\link{icd9cm_charlson_quan}},
#' \code{\link{melt_icd9list}}
#' @keywords datasets
#' @examples
#' charlson_weights_orig["aids"]
NULL

#' Map of Charlson comorbidity categories to revised weights
#' 
#' List that links the Charlson comorbidity categories to revised 
#' weights as calculated by Schneeweiss in Table 4 of his paper.
#' 
#' Revised Schneeweiss weights:
#' 
#' 0 = Connective tissue dz, Ulcer
#' 
#' 1 = MI, PVD, CVD, Diabetes, Hemiplegia
#' 
#' 2 = CHF, Chronic pulm dz, Mild liver dz, Diabetes with end organ damage, Any 
#' tumor, Leukemia, Lymphoma
#' 
#' 3 = Dementia, Mod or severe renal dz
#' 
#' 4 = Moderate or severe liver dz, AIDS
#' 
#' 6 = Metastatic solid tumor
#' 
#' @docType data
#' @name charlson_weights
#' @usage charlson_weights
#' @format A list, with Charlson comorbidities as names and weight as value
#' @references 1. Schneeweiss S, Wang PS, Avorn J, Glynn RJ: Improved comorbidity adjustment 
#'   for predicting mortality
#'   in Medicare populations. Health services research 2003; 38:1103
#'   \url{http://www.ncbi.nlm.nih.gov/pubmed/12968819}
#' @seealso \code{\link{charlson_weights_orig}},
#' \code{\link{icd9cm_charlson_deyo}}, 
#' \code{\link{icd9cm_charlson_romano}},
#' \code{\link{icd9cm_charlson_quan}},
#' \code{\link{melt_icd9list}}
#' @keywords datasets
#' @examples
#' charlson_weights["dementia"]
NULL

#' First 100 patients and their ICD-9-CM codes from the Vermont Uniform Hospital
#' Discharge Data Set for 2011, Inpatient.
#' 
#' Diagnostic ICD-9 codes are prefixed with 'D', while procedural ICD-9 codes are prefixed with 'P'.
#' 
#' @docType data
#' @name vt_inp_sample
#' @usage vt_inp_sample
#' @format A data frame, with column "id" (numeric), some descriptive columns,
#'   "dx" (factor), and "icd9cm" (factor)
#' @references 
#' \url{http://healthvermont.gov/research/hospital-utilization/RECENT_PU_FILES.aspx}
#' @seealso \code{\link{icd9cm_charlson_deyo}}, 
#' \code{\link{icd9cm_charlson_quan}}, 
#' \code{\link{icd9cm_charlson_romano}}, 
#' \code{\link{icd9cm_elixhauser_quan}}, 
#' \code{\link{icd9cm_elixhauser_ahrq37}}
#' @keywords datasets
#' @examples
#' max(vt_inp_sample$scu_days)
NULL

#' Values for calculating RSI for in-hospital mortality.
#' @docType data
#' @name rsi_beta_inhosp
#' @usage rsi_beta_inhosp
#' @format A hash (see package "hash"), where key is icd9cm code, and value is beta.
#' Special key "popbeta" has the population beta for the entire table.
#' @references 1. Sessler DI, Sigl JC, Manberg PJ, Kelley SD, Schubert A, Chamoun NG.
#'     Broadly applicable risk stratification system for predicting duration of hospitalization and mortality.
#'     Anesthesiology. 2010 Nov;113(5):1026-37. doi: 10.1097/ALN.0b013e3181f79a8d.
#' \url{http://my.clevelandclinic.org/anesthesiology/outcomes-research/risk-stratification-index.aspx}
NULL

#' Values for calculating RSI for 30-day mortality
#' @docType data
#' @name rsi_beta_30dpod
#' @usage rsi_beta_30dpod
#' @format A hash (see package "hash"), where key is icd9cm code, and value is beta.
#' Special key "popbeta" has the population beta for the entire table.
#' @references 1. Sessler DI, Sigl JC, Manberg PJ, Kelley SD, Schubert A, Chamoun NG.
#'     Broadly applicable risk stratification system for predicting duration of hospitalization and mortality.
#'     Anesthesiology. 2010 Nov;113(5):1026-37. doi: 10.1097/ALN.0b013e3181f79a8d.
#' \url{http://my.clevelandclinic.org/anesthesiology/outcomes-research/risk-stratification-index.aspx}
NULL

#' Values for calculating RSI for 1 year mortality
#' @docType data
#' @name rsi_beta_1yrpod
#' @usage rsi_beta_1yrpod
#' @format A hash (see package "hash"), where key is icd9cm code, and value is beta.
#' Special key "popbeta" has the population beta for the entire table.
#' @references 1. Sessler DI, Sigl JC, Manberg PJ, Kelley SD, Schubert A, Chamoun NG.
#'     Broadly applicable risk stratification system for predicting duration of hospitalization and mortality.
#'     Anesthesiology. 2010 Nov;113(5):1026-37. doi: 10.1097/ALN.0b013e3181f79a8d.
#' \url{http://my.clevelandclinic.org/anesthesiology/outcomes-research/risk-stratification-index.aspx}
NULL

#' Values for calculating RSI for 30-day length of stay
#' @docType data
#' @name rsi_beta_30dlos
#' @usage rsi_beta_30dlos
#' @format A hash (see package "hash"), where key is icd9cm code, and value is beta.
#' Special key "popbeta" has the population beta for the entire table.
#' @references 1. Sessler DI, Sigl JC, Manberg PJ, Kelley SD, Schubert A, Chamoun NG.
#'     Broadly applicable risk stratification system for predicting duration of hospitalization and mortality.
#'     Anesthesiology. 2010 Nov;113(5):1026-37. doi: 10.1097/ALN.0b013e3181f79a8d.
#' \url{http://my.clevelandclinic.org/anesthesiology/outcomes-research/risk-stratification-index.aspx}
NULL

#' Sample data for validating RSI
#' @docType data
#' @name rsi_sample_data
#' @usage rsi_sample_data
#' @seealso \code{\link{rsi_sample_results}}, \code{\link{verify_sessler_rsi}}
#' @format A data table with a patient ID and several columns with ICD-9-CM codes.
#' @references 1. Sessler DI, Sigl JC, Manberg PJ, Kelley SD, Schubert A, Chamoun NG.
#'     Broadly applicable risk stratification system for predicting duration of hospitalization and mortality.
#'     Anesthesiology. 2010 Nov;113(5):1026-37. doi: 10.1097/ALN.0b013e3181f79a8d.
#' \url{http://my.clevelandclinic.org/anesthesiology/outcomes-research/risk-stratification-index.aspx}
NULL

#' Sample results for validating RSI
#' @docType data
#' @name rsi_sample_results
#' @usage rsi_sample_results
#' @seealso \code{\link{rsi_sample_data}}, \code{\link{verify_sessler_rsi}}
#' @format A data table with a patient ID, principal diagnosis, principal procedure, and RSI result columns.
#' @references 1. Sessler DI, Sigl JC, Manberg PJ, Kelley SD, Schubert A, Chamoun NG.
#'     Broadly applicable risk stratification system for predicting duration of hospitalization and mortality.
#'     Anesthesiology. 2010 Nov;113(5):1026-37. doi: 10.1097/ALN.0b013e3181f79a8d.
#' \url{http://my.clevelandclinic.org/anesthesiology/outcomes-research/risk-stratification-index.aspx}
NULL
